gateway/service/src/main/java/org/eclipse/ditto/gateway/service/endpoints/actors/CheckPermissionsActor.java

```diff
@@ -0,0 +1,417 @@
+                .thenAccept(result -> processPolicyCommandResult(result, aggregateEntry.getValue(), permissionResults))
+                .exceptionally(ex -> {
+                    logError(ex, aggregateEntry.getKey().toString(), aggregateEntry.getValue().toString(),
+                            command.getDittoHeaders());
+
+                    throw DittoInternalErrorException.newBuilder()
+                            .dittoHeaders(command.getDittoHeaders())
+                            .cause(ex)
+                            .message("Failed to check permissions for policy ID: " + aggregateEntry.getKey())
+                            .build();
+                });
+    }
+
+    /**
+     * Processes the result of a policy command by updating the permission results.
+     *
+     * @param result the result of the policy command.
+     * @param resourcePermissions the resource permissions to check.
+     * @param permissionResults the map to store the permission check results.
+     */
+    private void processPolicyCommandResult(final Object result,
+            final Map<String, ResourcePermissions> resourcePermissions,
+            final Map<String, Boolean> permissionResults) {
+        if (result instanceof CheckPolicyPermissionsResponse response) {
+            permissionResults.putAll(CheckPolicyPermissionsResponse.toMap(response.getPermissionsResults()));
+        } else {
+            setPermissionCheckFailure(resourcePermissions, permissionResults);
+        }
+    }
+
+    /**
+     * Marks the permission check as failed by setting all results for the given resources to {@code false}.
+     *
+     * @param resourcePermissions the map of resource permissions that failed.
+     * @param permissionResults the map of permission results where the failure should be reflected.
+     */
+    private void setPermissionCheckFailure(Map<String, ResourcePermissions> resourcePermissions,
+            Map<String, Boolean> permissionResults) {
+        resourcePermissions.forEach((key, value) -> permissionResults.put(key, false));
+    }
+
+    /**
+     * Handles a failure during the permission checking process by logging the error and sending a
+     * {@link DittoInternalErrorException} back to the sender.
+     *
+     * @param ex the exception that occurred during the permission check.
+     * @param command the {@link CheckPermissions} command containing permission checks.
+     * @return null, as the error is propagated via an exception and response is sent.
+     */
+    private Void handleFailure(Throwable ex, CheckPermissions command) {
+        logError(ex, command.getPermissionChecks(), command.getDittoHeaders());
+
+        final DittoInternalErrorException errorResponse = DittoInternalErrorException.newBuilder()
+                .dittoHeaders(command.getDittoHeaders())
+                .cause(ex)
+                .message("Internal error while processing permission checks for command: " + command)
```

Comment:
The `toString` of `CheckPermissions` command is too long and technical for an exception message - this would e.g. be shown at the HTTP API as response.
I would not provide a message at all - the standard message should be sufficient for an internal server error.